### DISCUSSION TITLE
Passing size by props for dynamic tailwind class in react

### DISCUSSION THREAD

**Discussion by morphiezy** [tailwindlabs/tailwindcss#10114] [Help] [answered]
I have component `Avatar` in my react app : 

```jsx
const Avatar = ({size, url, caption }) => {
  return (
    <div
      className={`flex items-center justify-center w-[${size}px] h-[${size}px] cursor-pointer overflow-hidden rounded-full`}
    >
      // code
    </div>
  );
};
```

And when i try passing size to my component 

```js
<Avatar
    size={35}
    caption="avatar"
    url="avatar-url..."
 />
```

Tailwind not automatically generated class from this ultility class 

![Annotation 2022-12-18 153906](https://user-images.githubusercontent.com/71913929/208289014-f739bce6-9e59-442d-a59e-f99b294c5337.png)

Can anyone tell me what happen ? and how to fix it ?

**Comment by zslabs** [comment, marked as answer]
https://tailwindcss.com/docs/content-configuration#dynamic-class-names

**Reply by morphiezy** [reply]
Is it by using *complete class name* will solve the problem in my code ? but how do i apply it to my code? I'm still confused

**Reply by wongjn** [reply]
You would need to write the class name in full, like `w-[35px]` and `h-[35px]` instead of `w-[${size}px]` and `w-[${size}px]` respectively. You lose the ability to affect these properties via the `size` prop, so you could instead apply the properties via the `style` attribute.

**Reply by morphiezy** [reply]
Thanks @wongjn it's work for me...
But can this solution be used as a best practice for the same case in future projects?

**Reply by wongjn** [reply]
Yep!

**Reply by morphiezy** [reply]
Thanks @wongjn

**Comment by RamonMF1** [comment]
Awesome response, helped me when having this issue

**Comment by MFRamon** [comment]
had this same error and this answer solved it :)
